src/options.c

```diff
@@ -1352,26 +1352,25 @@ char *op;
         return optn_ok;
     }
     if (req == do_set) {
-        /* gender:string */
+        /* orientation:string */
         if (parse_role_opts(optidx, negated, allopt[optidx].name, opts, &op)) {
             if ((flags.initorientation = str2orientation(op)) == ROLE_NONE) {
                 config_error_add("Unknown %s '%s'", allopt[optidx].name, op);
+		flags.orientation = flags.initorientation = SEX_STRAIGHT;
                 return optn_err;
             } else
                 flags.orientation = flags.initorientation;
         } else {
             /* TODO: Find way to make straight not the default. */
-            flags.orientation = SEX_STRAIGHT;
+            flags.orientation = flags.initorientation = SEX_STRAIGHT;
```

Comment:
I had it explicitly set `flags.initorientation` when defaulting to `SEX_STRAIGHT`, so that the value displayed in the options menu will match your starting orientation even if some sort of \`\`amulet of conversion therapy'' is added to the game. I can't think of a reason you'd need to distinguish between a player deliberately setting `OPTIONS=orientation:straight` vs ignoring the option and falling back to the default -- but if there is some case where that'd be necessary, this is probably a bad idea?

The old way of doing it does set `flags.initorientation` when the player has set an invalid/impossible-to-parse orientation, but not if the player omits the option altogether.